lightningd/invoice.c

```diff
@@ -382,11 +383,12 @@ invoice_check_payment(const tal_t *ctx,
 	 *    - MUST return an `incorrect_or_unknown_payment_details` error.
 	 */
 	if (!invoices_find_unpaid(ld->wallet->invoices, &inv_dbid, payment_hash)) {
-		log_debug(ld->log, "Unknown paid invoice %s",
-			  type_to_string(tmpctx, struct sha256, payment_hash));
 		if (invoices_find_by_rhash(ld->wallet->invoices, &inv_dbid, payment_hash)) {
-			log_debug(ld->log, "ALREADY paid invoice %s",
-				  type_to_string(tmpctx, struct sha256, payment_hash));
+			*err = tal_fmt(ctx, "Already paid or expired invoice %s",
+				       type_to_string(tmpctx, struct sha256, payment_hash));
+		} else {
+			*err = tal_fmt(ctx, "Unknown invoice %s",
+				       type_to_string(tmpctx, struct sha256, payment_hash));
 		}
```

Comment:
we can write the if without `{...}` here just to be consistent with the our code

Reply:
No, if there is multiline code I prefer {} for clarity.  It's fine to omit for simple cases, ofc.